### PR TITLE
Add documentation about package signing model

### PR DESCRIPTION
**Pull request by kotfic**
This PR provides documentation for the current AWS based package signing approach and serves as part of a larger data integrity assurance case.

### CODE REVIEW

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst]: `SIgning` -> `Signing`

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst]: `issues` -> `issue`

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst]: Does this `TODO` refer to the table below?  In that case can it be removed.  Otherwise, what table is planned here?

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst]: I think we decided this was not needed.  But either way it's not implemented so I think removing this line and the one above makes sense.

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst]: I think we should borrow from the nice description @tgamblin wrote for the buildcache PR (#30750).  Maybe the layout he provided under "New Format" in the PR description could just be reused here.

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst:295]: Is this to be left `TODO` for the time being or completed before merging?

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst]: Missing a second backtic after `spack-intermediate-ci-signing-key`?

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst:324]: Suggest backtics around `spack-intermediate-ci-signing-key` here.

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst]: Second backtic missing after `aws-encryption-cli`?

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst]: `there protected` -> `there are protected`?

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst]: We now push the branches from github directly to gitlab without first adding the `github/` prefix, so these branch names should be updated accordingly.

Additionally, we run pipelines protected tags, also defined with wildards in gitlab, to match the tag format spack uses (i.e. `v*`).

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst]: `secrets its` -> `secrets, it's`

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst]: `the other` -> `untrusted code`?

File: lib/spack/docs/signing.rst
Comment (scottwittenburg) [inline lib/spack/docs/signing.rst]: `originated from` -> `originated,` ?

**Review by scottwittenburg — CHANGES_REQUESTED** [review]
Thanks @kotfic this is looking really great :tada: 

I've requested some minor changes, but the main issue might be that the docs build is failing.  There are a few places where double backtics are not closed, which might account for most of the doc build errors.  Also, this needs to be linked from somewhere in the table of contents or somewhere doesn't it? I think that's causing the final error.

Doc build errors:

```
/home/docs/checkouts/readthedocs.org/user_builds/spack/checkouts/30939/lib/spack/docs/signing.rst:307: WARNING: Inline literal start-string without end-string.
/home/docs/checkouts/readthedocs.org/user_builds/spack/checkouts/30939/lib/spack/docs/signing.rst:349: WARNING: Inline literal start-string without end-string.
/home/docs/checkouts/readthedocs.org/user_builds/spack/checkouts/30939/lib/spack/docs/signing.rst:427: WARNING: Inline literal start-string without end-string.
/home/docs/checkouts/readthedocs.org/user_builds/spack/checkouts/30939/lib/spack/docs/signing.rst:427: WARNING: Inline literal start-string without end-string.
/home/docs/checkouts/readthedocs.org/user_builds/spack/checkouts/30939/lib/spack/docs/signing.rst:427: WARNING: Inline literal start-string without end-string.
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/docs/checkouts/readthedocs.org/user_builds/spack/checkouts/30939/lib/spack/docs/signing.rst: WARNING: document isn't included in any toctree
```

File: lib/spack/docs/signing.rst
Comment (kotfic) [reply lib/spack/docs/signing.rst]: @scottwittenburg could you push a commit with that change?  i'm not as familiar with that PR or this section of the signing docs

**Comment by kotfic** [comment]
This appears to be passing now except for a warning about not being included in any toctree. @tgamblin Where would you like this documentation linked from?  Thanks!

**Review by scottwittenburg — DISMISSED** [review]

**Comment by alalazo** [comment]
Is this PR just waiting for another review?

**Comment by scottwittenburg** [comment]
> Is this PR just waiting for another review?

Also appears from a [comment](https://github.com/spack/spack/pull/30939#issuecomment-1146063367) above that @kotfic wanted to know where this should be linked from.

**Comment by alalazo** [comment]
What about inserting it below the sections on CI pipelines? That seems the appropriate place to me since we would introduce CI pipelines first, and then explain how we are signing the buildcaches from our public pipelines. Does that seem fine?

**Comment by scottwittenburg** [comment]
> What about inserting it below the sections on CI pipelines?

Yes, that makes sense to me.

**Comment by alalazo** [comment]
@kotfic If you have time to push this forward, I'll try to prioritize review to it.

**Comment by alalazo** [comment]
@scottwittenburg @zackgalbreath Can you double check if what is written here is still up to date? If so I'll take care of fixing the readthedocs failure and merge the PR.

**Comment by scottwittenburg** [comment]
I just read through it again, and yes, this is still an accurate and complete description of our package signing model.

**Comment by alalazo** [comment]
Thanks @scottwittenburg I'll try to merge this one asap then

**Review by alalazo — APPROVED** [review]

**Pull request merged**
